weather_data_analyzer.py

```diff
@@ -0,0 +1,198 @@
+        :param record:
+        :return: bool
+        """
+
+        if record.max_temp:
+            if int(max_high_temp.max_temp) < int(record.max_temp):
+                return True
+        return False
+
+    def low_temp_greater_than_record(self, min_low_temp, record):
+        """
+        Returns true if the given record has min_temp minimum than the already
+        min_temp of min_low_temp
+
+        :param min_low_temp:
+        :param record:
+        :return: bool
+        """
+
+        if record.min_temp:
+            if int(min_low_temp.min_temp) > int(record.min_temp):
+                return True
+        return False
+
+    def high_humid_less_than_record(self, max_high_humid, record):
+        """
+        Returns true if the given record has max_humidity greater than the
+        already max_humidity of max_high_humid
+
+        :param max_high_humid:
+        :param record:
+        :return: bool
+        """
+
+        if record.max_humidity:
+            if int(max_high_humid.max_humidity) < int(record.max_humidity):
+                return True
+        return False
+
+    def calc_average_temp_and_humidity(self):
+        """
+        Returns the average of max_temperature, min_temperature and
+        mean_humidity
+
+        :return: AverageTempHumidReportType
+        """
+
+        if len(self.file_records) == 0:
+            return None
+
+        report = AverageTempHumidReportType(
+            average_max_temperature=self.get_average_max_temperature(),
+            average_min_temperature=self.get_average_min_temperature(),
+            average_mean_humidity=self.get_average_mean_humidity()
+        )
+
+        return report
+
+    def get_average_max_temperature(self):
+        """
+        Calculates the average of max_temperature from the file_records list
+
+        :return: average - float
+        """
+
+        total_sum_of_max_temp = 0
+        for record in self.file_records:
+
+            if record.max_temp:
+                total_sum_of_max_temp += int(record.max_temp)
+
+        return total_sum_of_max_temp / len(self.file_records)
+
+    def get_average_min_temperature(self):
+        """
+        Calculates the average of min_temperature from the file_records list
+
+        :return: average - float
+        """
+
+        total_sum_of_min_temp = 0
+        for record in self.file_records:
+
+            if record.min_temp:
+                total_sum_of_min_temp += int(record.min_temp)
+
+        return total_sum_of_min_temp / len(self.file_records)
+
+    def get_average_mean_humidity(self):
+        """
+        Calculates the average of mean_humidity from the file_records list
+
+        :return: average - float
+        """
+
+        total_sum_of_mean_humidity = 0
+
+        for record in self.file_records:
+
+            if record.mean_humidity:
+                total_sum_of_mean_humidity += int(record.mean_humidity)
+
+        return total_sum_of_mean_humidity / len(self.file_records)
```

Comment:
you can make it more pythonic using array comprehension

Reply:
Nice. Something like below: 
We can use the 'mean' from python lib statistics. but the problem is that max_temp in a record can be None instead of integer.

```python
       max_temps = [r.max_temp for r in self.file_records if r.max_temp]
       total = sum(max_temps)
       return total / len(self.file_records)
```

Reply:
Now it depends what you want to do with None records, either remove these values in data cleaning otherwise you can use `r.get('max_temp',0) or 0` to assign default values.